src/Autosuggest.js

```diff
@@ -18,6 +20,12 @@ export default class Autosuggest extends Component { // eslint-disable-line no-s
     id: PropTypes.string                    // Used in aria-* attributes. If multiple Autosuggest's are rendered on a page, they must have unique ids.
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (typeof nextProps.value === 'string') {
+      this.onUpdate(nextProps.value);
```

Comment:
Well, this function is still sort of a handler function, because it happens in response to the `value` changing *or* the uncontrolled input text changing due to user typing. But, it is a purely-internal handler function, and not one that can be passed in via props, so perhaps a different naming scheme does make sense.

What do you think about `handleValueChange` as a name?